### PR TITLE
[doc] Update notary SPN roles for the publisher

### PR DESCRIPTION
**Pull request by ggjjj**
AcrImageSigner role is missing in the document. Updated that and also the Notary's latest version to 0.6.1

### CODE REVIEW

File: doc/NotaryContentTrust.md [hunk 1]
```diff
@@ -91,7 +91,7 @@ Notary client directory exists in `~/.notary`. A configuration file has be creat
...
+Notary needs authorization credentials to communicate to the Notary Server. For this, a Service Principal for the Azure Container Registry with Owner, AcrImageSigner and AcrPush access must be created. Follow instructions in [this](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-auth-service-principal) link to create a Service Principal. It is mandatory to create the Service Principals to use the Content Trust feature of Azure Container Registry. Read [this](https://docs.microsoft.com/en-us/azure/iot-edge/production-checklist?view=iotedge-2020-11#security-considerations) documentation of IoT Edge on how to use Service Principals.
```
Comment (damonbarry) [inline doc/NotaryContentTrust.md:94]: If the SP has AcrImageSigner and AcrPush rights, does it still need Owner?

Comment (ggjjj) [reply doc/NotaryContentTrust.md:94]: Yes In one of the steps to to rotate keys, it was failing. Once I added this role - it was successful.

Comment (ggjjj) [reply doc/NotaryContentTrust.md:94]: Owner role for setting the trust collection, signer role for key rotation and Push role for pushing image to repo

Comment (damonbarry) [reply doc/NotaryContentTrust.md:94]: Owner gives the SP rights to do _anything_, including assign other roles, so it's not a secure option. I'm wondering if the notary client can set up the TUF collection with a lesser access role? Have you tried granting _just_ AcrImageSigner and AcrPush (not Owner)? [Some documentation](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-content-trust#grant-image-signing-permissions) I found only mentioned AcrImageSigner and AcrPush.

Comment (ggjjj) [reply doc/NotaryContentTrust.md:94]: Let me try without Owner.

Comment (ggjjj) [reply doc/NotaryContentTrust.md:94]: I tried without the owner role and while notary key rotation - it says the following error

Comment (damonbarry) [reply doc/NotaryContentTrust.md:94]: > For making any changes in the signature collection, Owner role is essential

Did you write that, or was it part of the error message?

If that's not part of the error message, then please try with at least Contributor instead of Owner. If that doesn't work, we can keep Owner in the doc, but I'm hoping we don't have to.

**Comment by darobs** [comment]
Lines 67-68:

> 
> - The Comman Name (CN) in root certificate for each container must contain the Globally Unique Names(GUNs) i.e exampleregistry.azurecr.io/image 
> -  Safely store the pass code for the private key for `root_ca_exampleregistry.key` and `root_image.key` and also it will be used in the next steps. 

67: spelling error (should be Common).
68: I think a better wording might be: Safely store the pass code for the private key for `root_ca_exampleregistry.key` and `root_image.key`. These will be used in the next steps.

File: doc/NotaryContentTrust.md [hunk 2]
```diff
@@ -73,7 +73,7 @@ Notary initializes the TUF trust collection for each image in the Container Regi
 ##### a. Notary Installation
 Notary client can be installed with the command for `amd64` target platform and avoid installing using `sudo apt-get install notary`
```
Comment (darobs) [inline doc/NotaryContentTrust.md:74]: Notary can be installed with the following wget command for `amd64` target platform.  Avoid installing using `apt-get`.  
(Why should we avoid installing via apt?)

**Review by darobs — COMMENTED** [review]
Lines 130-260 look like a copy of lines 1-130 is there a copy/paste error here?

File: doc/NotaryContentTrust.md (hunk 1 above)
Comment (darobs) [reply doc/NotaryContentTrust.md:94]: If we only need Owner/Contributor for key rotation, perhaps we should tell the user to create 3 service principals for each role in the process, Owner for Step 2, SignImage/Push for Step 3. Pull for Step 4?

Comment (ggjjj) [reply doc/NotaryContentTrust.md:94]: > > For making any changes in the signature collection, Owner role is essential
> 
> Did you write that, or was it part of the error message?
> 
> If that's not part of the error message, then please try with at least Contributor instead of Owner. If that doesn't work, we can keep Owner in the doc, but I'm hoping we don't have to.

I did write that and it was not part of the message. I tried with multiple service principals. Here is the result
1. Only owner - Fails to initialize the trust collection i.e. Notary init fails
2. Owner, AcrPush - Key rotate fails but can initialize the trust collection. 
3. AcrPush, AcrImageSigner - Key rotate fails but can initialize the trust collection. 
4. Owner, AcrPush, AcrImageSigner - Can initialize and also rotate the key. 
5. Owner, AcrImageSigner - cannot initialize the trust collection

So in conclusion, we need all three as minimum to do the operation. @darobs I am not sure which role plays what so I will leave only the generic instructions on which roles are needed to get started.

File: doc/NotaryContentTrust.md (hunk 2 above)
Comment (ggjjj) [reply doc/NotaryContentTrust.md:74]: ACR notary contact mentioned that the list is not update regularly and we would want a specific version. so this method.

Comment (ggjjj) [reply doc/NotaryContentTrust.md:74]: I noticed the copy paste. Will remove that

File: doc/NotaryContentTrust.md (hunk 1 above)
Comment (darobs) [reply doc/NotaryContentTrust.md:94]: @ggjjj Were you able to assign & use Contributor?

Comment (darobs) [reply doc/NotaryContentTrust.md:94]: RE: the 3 service principals.  I was hoping to model each one as the roles needed to use this feature.  We would expect the first user (a person as an admin/devops) to have the ability to install certificates and rotate keys - they would have Owner, AcrPush, AcrImageSigner permissions.  The second user is an image builder (probably automated) that would be the one that pushes and signs the image - they would have AcrPush, AcrImageSigner permissions. Third, there would be a user of that image (like IoT Edge) that pulls it.  This may be slightly out of scope for this document, but if it doesn't muck up the flow too much, it would be a nice addition.

Comment (ggjjj) [reply doc/NotaryContentTrust.md:94]: Contributor, AcrPush, AcrImageSigner works

**Review by damonbarry — COMMENTED** [review]
Looks good to me. @darobs?

**Comment by darobs** [comment]
> Looks good to me. @darobs?

I have a Github tab somewhere where I forgot to hit Approve...

**Review by darobs — APPROVED** [review]
